apps/communications/dialer/test/marionette/add_contact_from_dialer.js

```diff
@@ -0,0 +1,38 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('chai').assert;
+
+var Dialer = require('./lib/dialer.js');
+
+marionette('Dialer > Keypad', function() {
+
+  var client = marionette.client({
+    desiredCapabilities: { raisesAccessibilityExceptions: false }
+  });
+
+  var dialerApp;
+
+  setup(function() {
+  
+    dialerApp = new Dialer(client);
+    dialerApp.launch();
+  
+   });
+
+  suite('Test Suite', function() {
+
+    test('Add a contact from dialer', function() {
+        
+        var typeNumber = dialerApp.keypad.typePhoneNumber(12345);
+        var addAsNewContact = typeNumber.addToContacts();
+        var contact = addAsNewContact.tapCreateContactOption();
+        var dialerOptions = contact.tapSavefromDialer();
+        var contactList = dialerOptions.goToContactList();
+        var contactCreated = contactList.checkContactCreated();
+        
+        assert.ok(contactCreated === 'Name', 'Contact is in contact list');
```

Comment:
That assertion can be improved, by using [assert.strictEqual](http://chaijs.com/api/assert/#-notequal-actual-expected-message-).

Also the custom error message [will hide valuable debug data](https://wiki.mozilla.org/B2G/QA/Automation/Style_Guide/Best_Practices#Assertion_Messages).